fuel-core-interfaces/src/db.rs

```diff
@@ -0,0 +1,545 @@
+use std::io::ErrorKind;
+
+use fuel_vm::prelude::InterpreterError;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("error performing binary serialization")]
+    Codec,
+    #[error("Failed to initialize chain")]
+    ChainAlreadyInitialized,
+    #[error("Chain is not yet initialized")]
+    ChainUninitialized,
+    #[error("Invalid database version")]
+    InvalidDatabaseVersion,
+    #[error("error occurred in the underlying datastore `{0}`")]
+    DatabaseError(Box<dyn std::error::Error + Send + Sync>),
+}
+
+impl From<Error> for std::io::Error {
+    fn from(e: Error) -> Self {
+        std::io::Error::new(ErrorKind::Other, e)
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum KvStoreError {
+    #[error("generic error occurred")]
+    Error(Box<dyn std::error::Error + Send + Sync>),
+    #[error("resource not found")]
+    NotFound,
+}
+
+impl From<Error> for KvStoreError {
+    fn from(e: Error) -> Self {
+        KvStoreError::Error(Box::new(e))
+    }
+}
+
+impl From<KvStoreError> for Error {
+    fn from(e: KvStoreError) -> Self {
+        Error::DatabaseError(Box::new(e))
+    }
+}
+
+impl From<KvStoreError> for std::io::Error {
+    fn from(e: KvStoreError) -> Self {
+        std::io::Error::new(ErrorKind::Other, e)
+    }
+}
+
+impl From<Error> for InterpreterError {
+    fn from(e: Error) -> Self {
+        InterpreterError::Io(std::io::Error::new(std::io::ErrorKind::Other, e))
+    }
+}
+
+impl From<KvStoreError> for InterpreterError {
+    fn from(e: KvStoreError) -> Self {
+        InterpreterError::Io(std::io::Error::new(std::io::ErrorKind::Other, e))
+    }
+}
+
+#[cfg(any(test, feature = "test_helpers"))]
+pub mod helpers {
+
+    use core::str::FromStr;
+    use std::sync::Arc;
+
+    use fuel_asm::Opcode;
+    use fuel_storage::Storage;
+    use fuel_tx::{
+        Address, Bytes32, ContractId, Input, Metadata, Output, Transaction, TxId, UtxoId,
+    };
+    use fuel_vm::prelude::Contract;
+    use std::collections::{HashMap, HashSet};
+
+    use crate::txpool::TxPoolDB;
+
+    use super::*;
+    #[derive(Clone, Debug)]
+    pub struct DummyDB {
+        pub tx_hashes: Vec<TxId>,
+        pub tx: HashMap<TxId, Arc<Transaction>>,
+        pub contract: HashSet<ContractId>,
+    }
+
+    impl DummyDB {
+        ///
+        pub fn dummy_tx(txhash: TxId) -> Transaction {
+            // One transfer tx1 depends on db
+            // One dependent transfer tx2 on tx1
+            // One higher gas_price transfer tx3 from tx1
+            // one higher gas_price transfer tx4 then tx2
+            // tx5 that depends on tx1 contract
+            // tx6 same as tx1 but without coin output
+
+            let db1 = TxId::from_str(
+                "0x00000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
[nit] Could we use consts for these instead? The tests are pretty hard to read when trying to cross-reference with these tx id strings here.

Reply:
Done the refactoring, and used `lazy_static` to make them static, it is a lot nicer now.